llmfoundry/data/finetuning/collator.py

```diff
@@ -320,24 +327,33 @@ def _process_and_batch_decoder_only(
     ) -> Dict[str, torch.Tensor]:
         # Steps explained in comments
         processed_examples = []
-        for example in examples:
-            input_ids, labels = stitch_turns_decoder_only(
+        input_ids_and_labels = [
+            stitch_turns_decoder_only(
                 example_turns=example['turns'],
                 target_prompts=self.target_prompts,
                 target_responses=self.target_responses,
                 eos_token_id=self.tokenizer.eos_token_id,
-            )
+            ) for example in examples
+        ]
+
+        if self._pad_to_longest:
+            max_seq_len = max([
```

Comment:
You want to also include the true max seq len here
```suggestion
            max_seq_len = max([self.max_seq_len] + [
```

Reply:
oh my bad it should be 
```
max_seq_len = min(self.max_seq_len, max([...]))
```
right?

Reply:
I think the test was failing because the examples were not getting truncated

Reply:
Yea fixed it